metrics/nist/nist.py

```diff
@@ -0,0 +1,132 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""NLTK's NIST implementation on both the sentence and corpus level"""
+import datasets
+from datasets import Sequence, Value
+from nltk.translate.nist_score import corpus_nist, sentence_nist
+
+import evaluate
+
+
+_CITATION = """\
+@inproceedings{10.5555/1289189.1289273,
+    author = {Doddington, George},
+    title = {Automatic Evaluation of Machine Translation Quality Using N-Gram Co-Occurrence Statistics},
+    year = {2002},
+    publisher = {Morgan Kaufmann Publishers Inc.},
+    address = {San Francisco, CA, USA},
+    booktitle = {Proceedings of the Second International Conference on Human Language Technology Research},
+    pages = {138–145},
+    numpages = {8},
+    location = {San Diego, California},
+    series = {HLT '02}
+}
+"""
+
+_DESCRIPTION = """\
+    DARPA commissioned NIST to develop an MT evaluation facility based on the BLEU
+    score. The official script used by NIST to compute BLEU and NIST score is
+    mteval-14.pl. The main differences are:
+
+     - BLEU uses geometric mean of the ngram overlaps, NIST uses arithmetic mean.
+     - NIST has a different brevity penalty
+     - NIST score from mteval-14.pl has a self-contained tokenizer
+"""
+
+
+_KWARGS_DESCRIPTION = """
+Computes NIST score of translated segments against one or more references.
+Args:
+    predictions: tokenized predictions to score. For sentence-level NIST, a list of tokens (str);
+     for corpus-level NIST, a list (sentences) of lists of tokens (str)
+    references:  potentially multiple tokenized references for each prediction.  For sentence-level NIST, a
+     list (multiple potential references) of list of tokens (str); for corpus-level NIST, a list (corpus) of lists
+     (multiple potential references) of lists of tokens (str)
+    n: highest n-gram order
+Returns:
+    'nist': nist score 
+Examples:
+    >>> nist = evaluate.load("nist")
+    >>> hypothesis1 = ['It', 'is', 'a', 'guide', 'to', 'action', 'which',
+    ...               'ensures', 'that', 'the', 'military', 'always',
+    ...               'obeys', 'the', 'commands', 'of', 'the', 'party']
+
+    >>> reference1 = ['It', 'is', 'a', 'guide', 'to', 'action', 'that',
+    ...               'ensures', 'that', 'the', 'military', 'will', 'forever',
+    ...               'heed', 'Party', 'commands']
+
+    >>> reference2 = ['It', 'is', 'the', 'guiding', 'principle', 'which',
+    ...               'guarantees', 'the', 'military', 'forces', 'always',
+    ...               'being', 'under', 'the', 'command', 'of', 'the',
+    ...               'Party']
+
+    >>> reference3 = ['It', 'is', 'the', 'practical', 'guide', 'for', 'the',
+    ...               'army', 'always', 'to', 'heed', 'the', 'directions',
+    ...               'of', 'the', 'party']
```

Comment:
what's reference three doing here :)

Reply:
Ah, this is from the original NLTK examples. Will update the examples with sentence-level + add to tests.